README.rst

```diff
@@ -141,10 +143,13 @@ One may also use other element types::
   a = DisjointSets{AbstractString}(["a", "b", "c", "d"])
   union!(a, "a", "b")
   in_same_set(a, "c", "d")
+  root_union!(a, "a", "d")
   push!(a, "f")
 
 
-Note that the internal implementation of ``IntDisjointSets`` is based on vectors, and is very efficient. ``DisjointSets{T}`` is a wrapper of ``IntDisjointSets``, which uses a dictionary to map input elements to an internal index.
+Note that the internal implementation of ``IntDisjointSets`` is based on vectors, and is very efficient. ``DisjointSets{T}`` is a wrapper of ``IntDisjointSets``, which uses a dictionary to map input elements to an internal index. 
+As a result, ``union!(a, "a", "b")`` and ``root_union!(a, "a", "d")`` returns the indices of the
+root elements rather than the root elements themselves.
```

Comment:
> Is there a way to go from index to element?

Since ``DisjointSets`` use a dictionary to wrap ``IntDisjointSets``, "to go from index to element" means get a key by value in a dictionary. This is doable but not very efficient, perhaps something like
```julia
for (key, value) in intmap
     if value == index
          return key
     end
end
```
Perhaps we could add a new field to ``DisjointSets``, but this is a major change and shoud be considered in a different pull request. By the way, currently ``find_root`` only  returns the index of the root rather than the root itself.

> If someone is manipulating non-integer DisjointSets, is returning the resulting index is useful? Can they use that index in further processing? How?

Yes, the index can be used in further processing. Just use the simple example from the documentation. 
```julia
> list = ["a", "b", "c", "d"]
> a = DisjointSets{AbstractString}(list)
> indx = union!(a, "a", "b")
> list[indx]
"a"
```